### DISCUSSION TITLE
Both onion engines time out

### DISCUSSION THREAD

**Discussion by ghost** [searx/searx#3383] [Installation issue: custom]
I have installed tor and Searx on Arch Linux and I have followed the suggestions in this subreddit for routing the queries through tor proxy. However neither ahmia nor torch work, I got a timeout error whatever request timeout I set (i tried to extend it up to 20 secs). All the other engines seem to work though I can't be absolutely sure that routing through tor is actually happening. What could I do wrong? This is my relevant section in settings.yml (sorry but the correct indentation does not show up after I post) :
... 
outgoing:
   request_timeout: 10.0
   proxies:
       all://:
          - socks5h://127.0.0.1:9050
   using_tor_proxy: true
..

**Comment by ghost** [comment]
@MarcAbonce do you have any suggestion?

**Reply by ghost** [reply]
Just to add an info, onion searches work with SearxNG, but unfortunately I can't use it because httpx client crashes at boot.

**Reply by MarcAbonce** [reply]
Sorry, but I have no answer for this so far.
Today I tried to run some queries locally and sometimes I got a response on time (<10 seconds) and sometimes I got a timeout.
I'm not sure if this is just the engines being slow or if there's an issue with the way we're sending the requests.

**Reply by ghost** [reply]
Thanks for the reply, I am trying to get Searxng to work and i noticed that in network.py there is this line:
response = await client.get("https://check.torproject.org/api/ip", timeout=10)
I extended the timeout to 20 and after a few reboots it seems to work.
